### PR TITLE
feat: implement Compact for OpTxType

### LINKED ISSUE #12516: Implement `Compact` for `OpTxType`
### Describe the feature

Implement `reth_codecs::Compact` for `OpTxType`. Check out how it's done for `TxType`
https://github.com/paradigmxyz/reth/blob/9313737dbbf0f46fa497e63d452b427c7dafe498/crates/primitives/src/transaction/tx_type.rs#L137-L188

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by kamuik16**
Closes #12516 

cc @emhane

### CODE REVIEW

File: crates/optimism/primitives/src/op_tx_type.rs [hunk 1]
```diff
@@ -10,9 +11,27 @@ use derive_more::{
...
+/// Identifier parameter for EIP-1559 transaction
+#[cfg(any(test, feature = "reth-codec"))]
+pub(crate) const COMPACT_IDENTIFIER_EIP1559: usize = 2;
+
+/// For backwards compatibility purposes only 2 bits of the type are encoded in the identifier
+/// parameter. In the case of a [`COMPACT_EXTENDED_IDENTIFIER_FLAG`], the full transaction type is
+/// read from the buffer as a single byte.
+#[cfg(any(test, feature = "reth-codec"))]
+pub(crate) const COMPACT_EXTENDED_IDENTIFIER_FLAG: usize = 3;
+
```
Comment (emhane) [inline crates/optimism/primitives/src/op_tx_type.rs:34]: make these public in `reth-primitives` instead and import them

**Review by emhane — CHANGES_REQUESTED** [review]
thanks a lot !

File: crates/optimism/primitives/src/bedrock.rs [hunk 2]
```diff
@@ -43,12 +43,12 @@ pub const BLOCK_NUMS_REPLAYED_TX: [u64; 6] = [
 /// with replayed transaction happen to only contain the single transaction.
 pub fn is_dup_tx(block_number: u64) -> bool {
     if block_number > BLOCK_NUMS_REPLAYED_TX[5] {
-        return false
+        return false;
```
Comment (emhane) [inline crates/optimism/primitives/src/bedrock.rs:46]: pls revert unrelated formatting changes like this one

File: crates/primitives/src/lib.rs [hunk 3]
```diff
@@ -56,6 +56,12 @@ pub use transaction::{
     TransactionSignedEcRecovered, TransactionSignedNoHash, TxHashOrNumber, TxType,
 };
 
+#[cfg(feature = "reth-codec")]
+pub use transaction::{
+    COMPACT_EXTENDED_IDENTIFIER_FLAG, COMPACT_IDENTIFIER_EIP1559, COMPACT_IDENTIFIER_EIP2930,
+    COMPACT_IDENTIFIER_LEGACY,
+};
+
```
Comment (emhane) [inline crates/primitives/src/lib.rs:64]: let's remove this export, and only make these accessible with `tx_type` module in the path. this, since the constant names don't contain tx type, so it can be a bit confusing what they refer to.

**Review by emhane — CHANGES_REQUESTED** [review]
almost there!

File: crates/optimism/primitives/src/bedrock.rs (hunk 2 above)
Comment (kamuik16) [reply crates/optimism/primitives/src/bedrock.rs]: https://github.com/paradigmxyz/reth/pull/12537/commits/ad5150ea788ceb5aff86cbb3994db34a6e652c6f

File: crates/primitives/src/lib.rs (hunk 3 above)
Comment (kamuik16) [reply crates/primitives/src/lib.rs]: https://github.com/paradigmxyz/reth/pull/12537/commits/ad5150ea788ceb5aff86cbb3994db34a6e652c6f

**Review by emhane — APPROVED** [review]
lgtm
